Feature Add common films

Yarikov-MY opened the pull request.

Yarikov-MY committed:
feature «Общие фильмы».

anbananova commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -83,7 +86,7 @@ public List<Film> getPopularFilmsByLikes(final Integer numberOfFilms) {
 
         return filmStorage.getAllFilms().stream()
-                .sorted((film1, film2) -> (film1.getUsersLikes().size() - film2.getUsersLikes().size()) * -1)
+                .sorted(new FilmLikesComparator())
```
мне кажется, создание отдельного компаратора излишне. сортировка не такая сложная, чтобы делать для нее отдельный класс. Если нужен обратный порядок сортировки от большего к меньшего, можно использовать sorted.reversed

будет что-то типа такого:
```.sorted(Comparator.comparing(FIlm::getUsersLikes().size()).reversed())```

Yarikov-MY replied:
Поправил

anbananova commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -114,4 +117,19 @@ public void isValidFilm(Film film) {
+            return new ArrayList<>();
+        }
+        List<Film> commonFilms = filmStorage.getFilmsById(likesIntersection);
+        return commonFilms.stream().sorted(new FilmLikesComparator()).collect(Collectors.toList());
```
то же самое с использованием компаратора

Yarikov-MY replied:
Поправил

anbananova reviewed (changes requested):
все хорошо, можно сделать рефакторинг сортировки в стримах

JuliaOlshevskaia commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -114,4 +117,19 @@ public void isValidFilm(Film film) {
     }
+
+    public List<Film> getCommonFriendsFilms(final Integer userId, final Integer friendId) {
+        if (!userStorage.isUserExist(userId)) {
```
Можно добавить еще проверку на существование friendId

Yarikov-MY replied:
Поправил

Yarikov-MY committed:
fix:
- убрать отдельный класс comparator;
- добавить проверку на существование друга.

Yarikov-MY committed:
fix:
Resolve conflicts add-common-films branch with develop.
